tests/Type/Nette/StringsMatchDynamicReturnTypeExtensionTest.php

```diff
@@ -12,7 +13,11 @@ class StringsMatchDynamicReturnTypeExtensionTest extends TypeInferenceTestCase
 	 */
 	public function dataFileAsserts(): iterable
 	{
-		yield from $this->gatherAssertTypes(__DIR__ . '/data/strings-match.php');
+		if (PHP_VERSION_ID < 70400) {
+			return;
+		}
+
+		yield from self::gatherAssertTypes(__DIR__ . '/data/strings-match.php');
```

Comment:
Can you please explain what is it about these example that they cannot work on PHP 7.2 and 7.3?

Reply:
The ArrayShapeMatcher requires named capturing groups to work

https://github.com/phpstan/phpstan-src/blob/794b20390a4ee56341541948fcb829373b2af5ab/src/Type/Php/RegexArrayShapeMatcher.php#L53

Reply:
Can you link 3v4l.org example that shows different results for the cutoff versions? Something that's different between 7.3 and 7.4 and that makes this not work for these older versions.

I tried running the links in the comments but they have the same output https://3v4l.org/sOXbn https://3v4l.org/3SdDM

Reply:
See https://3v4l.org/dYdoO#v7.2.0